subnet/subnet.go

```diff
@@ -53,7 +53,7 @@ type Lease struct {
 }
 
 func (l *Lease) Key() string {
-	return MakeSubnetKey(l.Subnet)
+	return MakeSubnetKey(l.Subnet, &l.IPv6Subnet)
```

Comment:
Overall i made functions extended to now also take an *ip.IP6Net; So for code that's not ready to dual-stack yet you can just pass nil as golang has no option type. I can have a go at how it looks with passing ip.IP6Net by value

Reply:
It's fine, I would like to understand what's the benefit of using pointers in this case because maybe we should use it for ipv4 too in the ipv6-only flannel (in another PR of course), so please bear with me. In the case of non-dualStack, if passed by value, I guess the l.IPv6Subnet struct cannot be nil, right? So the code becomes a bit harder (e.g. initialize the struct, check if we are in the dualStack case, etc). Whereas if the pointer is nil, we can quickly conclude that it is not the dualStack case, am I wrong?

Reply:
You can go both ways; I think it works potentially a bit nicer out as value as it avoids checking for both nil *and* .Empty().. and avoids accidental null-pointer deferences